### PR TITLE
drivers: hwinfo: add NuMaker support for device id

### PR DESCRIPTION
**Pull request by ccli8**
This adds device id for Nuvoton NuMaker series.

### CODE REVIEW

File: drivers/hwinfo/hwinfo_numaker.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{
+	union {
+		uint32_t words[3];
+		uint8_t bytes[12];
+	} fmc_uid;
+
+	SYS_UnlockReg();
+	FMC_Open();
+
+	fmc_uid.words[0] = FMC_ReadUID(0);
```
Comment (alexanderwachter) [inline drivers/hwinfo/hwinfo_numaker.c:21]: Is the byteorder correct?

**Review by alexanderwachter — COMMENTED** [review]
Thanks for the addition.

**Review by alexanderwachter — APPROVED** [review]

File: drivers/hwinfo/hwinfo_numaker.c (hunk 1 above)
Comment (ccli8) [reply drivers/hwinfo/hwinfo_numaker.c]: fixed

**Review by cyliangtw — APPROVED** [review]
LGTM

**Pull request merged**